### PR TITLE
Support for RITA language

### PR DESCRIPTION
**Pull request by zaibacu**
Introduces Lexer for [RITA](https://github.com/zaibacu/rita-dsl)

### CODE REVIEW

File: pygments/lexers/rita.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    flags = re.MULTILINE | re.UNICODE
+
+    tokens = {
+        'root': [
+            (r'\n', Text),
+            (r'\s+', Text),
+            (r'#(.*?)\n', Comment.Single),
+            (r'@(.*?)\n', Operator),  # Yes, whole line as an operator
+            (r'(\"|\')(\w|\d|[^\"\']|(\\\")|(\\\'))+?(\"|\')', Literal),
```
Comment (birkenfeld) [inline pygments/lexers/rita.py:39]: It's probably better to split this into separate rules for single and double quoted strings; right now you can do `"blah'`...

Also, no empty strings?

File: pygments/lexers/rita.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    inherit, words
+from pygments.token import Comment, Operator, Keyword, Name, Literal, Punctuation, Text
+
+__all__ = ['RitaLexer']
+
+
+class RitaLexer(RegexLexer):
+    """
+    Lexer for `RITA <https://github.com/zaibacu/rita-dsl>`_
+     .. versionadded:: 2.7
```
Comment (birkenfeld) [inline pygments/lexers/rita.py:24]: rst requires an empty line before the versionadded directive.

File: pygments/lexers/rita.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    tokens = {
+        'root': [
+            (r'\n', Text),
+            (r'\s+', Text),
+            (r'#(.*?)\n', Comment.Single),
+            (r'@(.*?)\n', Operator),  # Yes, whole line as an operator
+            (r'(\"|\')(\w|\d|[^\"\']|(\\\")|(\\\'))+?(\"|\')', Literal),
+            (r'([A-Z_]+)', Keyword),
+            (r'([a-z_]+)', Name),
```
Comment (birkenfeld) [inline pygments/lexers/rita.py:41]: No digits allowed in names?

**Review by birkenfeld — COMMENTED** [review]
Thanks for the PR! I've commented a few things to discuss.

File: pygments/lexers/rita.py (hunk 1 above)
Comment (zaibacu) [reply pygments/lexers/rita.py]: Thank you, that's a fair point and a nice way to solve it

**Comment by zaibacu** [comment]
Some times one of the builds freezes. Any ideas?

**Comment by Anteru** [comment]
I have to guess but I believe it's a randomized test hitting a regex which causes catastrophic backtracking.

**Comment by zaibacu** [comment]
> I have to guess but I believe it's a randomized test hitting a regex which causes catastrophic backtracking. Can you try to run it locally with the same random seed: `--randomly-seed=2266720041` and see if you get the same hang?

Yes, thank you, can reproduce it with this seed and indeed it crashes on `RitaLexer`. Looking how to fix regex so it would not hang

**Comment by zaibacu** [comment]
Bump :) this one is still not merged

**Pull request merged**